yb-voyager/cmd/endMigrationCommand.go

```diff
@@ -0,0 +1,418 @@
+	tdb := tgtdb.NewTargetDB(msr.TargetDBConf)
+	err := tdb.Init()
+	if err != nil {
+		utils.ErrExit("end migration: initializing target db: %v", err)
+	}
+	defer tdb.Finalize()
+	err = tdb.ClearMigrationState(migrationUUID, exportDir)
+	if err != nil {
+		utils.PrintAndLog("end migration: clearing migration state from target db: %v", err)
+	}
+}
+
+func cleanupFallForwardDB(msr *metadb.MigrationStatusRecord) {
+	if !msr.FallForwardEnabled {
+		return
+	}
+
+	utils.PrintAndLog("cleaning up fall-forward db...")
+	ffdb := tgtdb.NewTargetDB(msr.FallForwardDBConf)
+	err := ffdb.Init()
+	if err != nil {
+		utils.ErrExit("end migration: initializing fallforward db: %v", err)
+	}
+	defer ffdb.Finalize()
+	err = ffdb.ClearMigrationState(migrationUUID, exportDir)
+	if err != nil {
+		utils.PrintAndLog("end migration: clearing migration state from fallforward db: %v", err)
+	}
+}
+
+func cleanupFallBackDB(msr *metadb.MigrationStatusRecord) {
+	if !msr.FallbackEnabled {
+		return
+	}
+
+	utils.PrintAndLog("cleaning up fallback db...")
+	fbdb := tgtdb.NewTargetDB(msr.SourceDBAsTargetConf)
+	err := fbdb.Init()
+	if err != nil {
+		utils.ErrExit("end migration: initializing fallback db: %v", err)
+	}
+	defer fbdb.Finalize()
+	err = fbdb.ClearMigrationState(migrationUUID, exportDir)
+	if err != nil {
+		utils.PrintAndLog("end migration: clearing migration state from fallback db: %v", err)
+	}
+}
+
+func cleanupExportDir() {
+	utils.PrintAndLog("cleaning up export dir...")
+	subdirs := []string{"schema", "data", "logs", "reports", "temp", "metainfo"}
+	for _, subdir := range subdirs {
+		err := os.RemoveAll(filepath.Join(exportDir, subdir))
+		if err != nil {
+			utils.ErrExit("end migration: removing %s directory: %v", subdir, err)
+		}
+	}
+}
+
+func validateEndMigrationFlags(cmd *cobra.Command) error {
+	flags := []string{"backup-schema-files", "backup-data-files", "save-migration-reports", "backup-log-files"}
+	for _, flag := range flags {
+		if cmd.Flag(flag).Value.String() == "true" && !cmd.Flag("backup-dir").Changed {
+			return fmt.Errorf("flag %s requires --backup-dir flag to be set", flag)
+		}
+	}
+
+	if !utils.FileOrFolderExists(backupDir) {
+		return fmt.Errorf("backup-dir %q doesn't exists", backupDir)
+	}
+	return nil
+}
+
+func checkIfEndCommandCanBePerformed(msr *metadb.MigrationStatusRecord) {
+	// check if any ongoing voyager command
+	matches, err := filepath.Glob(filepath.Join(exportDir, ".*.lck"))
+	if err != nil {
+		utils.ErrExit("end migration: checking for ongoing voyager commands: %v", err)
+	}
+	if len(matches) > 0 {
+		var ongoingCmds []string
+		for _, match := range matches {
+			match = strings.TrimPrefix(match, ".")
+			match = strings.TrimSuffix(match, "Lockfile.lck")
+			if match == "end-migration" {
+				continue
+			}
+			ongoingCmds = append(ongoingCmds, match)
+		}
+		if len(ongoingCmds) > 0 &&
+			!utils.AskPrompt(fmt.Sprintf("found ongoing voyager commands: %s. Do you want to continue", strings.Join(ongoingCmds, ", "))) {
```

Comment:
Confusing prompt? If user types `yes` what will "continue" -- the "ongoing" commands or the current "end migration" command?